apiserver/facades/client/controller/controller.go

```diff
@@ -249,36 +253,21 @@ func (c *ControllerAPI) AllModels(ctx context.Context) (params.UserModelList, er
 		return result, errors.Trace(err)
 	}
 
-	modelUUIDs, err := c.state.AllModelUUIDs()
+	models, err := c.modelService.ListAllModels(ctx)
 	if err != nil {
 		return result, errors.Trace(err)
 	}
-	for _, modelUUID := range modelUUIDs {
-		st, err := c.statePool.Get(modelUUID)
-		if err != nil {
-			// This model could have been removed.
-			if errors.Is(err, errors.NotFound) {
-				continue
-			}
-			return result, errors.Trace(err)
-		}
-		defer st.Release()
-
-		model, err := st.Model()
-		if err != nil {
-			return result, errors.Trace(err)
-		}
-
+	for _, model := range models {
 		userModel := params.UserModel{
 			Model: params.Model{
-				Name:     model.Name(),
-				UUID:     model.UUID(),
-				Type:     string(model.Type()),
-				OwnerTag: model.Owner().String(),
+				Name:     model.Name,
+				UUID:     model.UUID.String(),
+				Type:     model.ModelType.String(),
+				OwnerTag: model.OwnerName.Name(),
 			},
 		}
 
-		lastConn, err := c.accessService.LastModelLogin(ctx, user.NameFromTag(c.apiUser), coremodel.UUID(model.UUID()))
+		lastConn, err := c.accessService.LastModelLogin(ctx, user.NameFromTag(c.apiUser), model.UUID)
```

Comment:
This PR didn't start this but we should really look to form up a new service method that supplies this information in one shot. This is very awkward at the moment.

I will make a card for us to do this work.

Reply:
Sure thing thanks tom!